index.js

```diff
@@ -1,52 +1,116 @@
-// all right our new element needs style. For that, assign to the new element the className 'square'!. Each time the user clicks on the new element it has to respond changing the displayed text 'x' to 'o'
-
-
-
+//Is time to create things in the DOM!!. We want you to create a new 'div' Element and
+//append it as a child of the element that has the 'id = a-1'(the element is in index.html file)
+// all right our new element needs style. For that, assign to the new element the className 'square'!.
+//Each time the user clicks on the new element it has to respond changing the displayed text 'x' to 'o'
 
 //resolve // QUESTION 1 here
+let newDiv = document.createElement("div"); //new div element is created
+let parentDiv1 = document.getElementById("a-1"); //get a div which is to be a parent for our new div
+parentDiv1.appendChild(newDiv); //new div element is appended to its parent
+newDiv.className = "square"; //a class name for the new div element is set
 
-
-
+//a listener for the click event is set on the new div element
+newDiv.addEventListener("click", function () {
+  //an action for the event is set: change of the inner html content via ternary operator
+  newDiv.innerHTML = newDiv.innerHTML === "x" ? "o" : "x";
+});
 
 //------------------------Question 2 ---------------------------
 //Write a JavaScript program to remove items from a dropdown list.
-//In order to create the list we are providing you with the array bellow, those items are the ones you need to add to the dropdown list as the select options
+//In order to create the list we are providing you with the array bellow,
+//those items are the ones you need to add to the dropdown list as the select options
 //you also need to create a button that makes the remove action of the element selected
 // append the new list as a child of the element with the 'id = a-2'
 // append the button as a child of the element with the 'id = a-2'
-const colors = [ 'red' , 'white', 'black' , 'green' , 'orange'];
-
-
-
 
 //resolve // QUESTION 2 here
-
-
-
-
-
+const colors = ["red", "white", "black", "green", "orange"];
+
+let parentDiv2 = document.getElementById("a-2");
+let dropdList = document.createElement("select"); //creating an element for dropdown list and assigning it to the newly declared variable
+let removeBtn = document.createElement("button"); //creating an element for a button and assigning it to the newly declared variable
+removeBtn.innerHTML = "Remove the option"; //button text
+//new elements are placed in DOM nested in the parent element
+parentDiv2.appendChild(dropdList);
+parentDiv2.appendChild(removeBtn);
+
+//creting the dropdown list
+for (var i = 0; i < colors.length; i++) {
+  let option = document.createElement("option");
+  option.value = colors[i];
+  option.text = colors[i];
+  dropdList.appendChild(option);
+}
+//function for removing the selected option
+removeBtn.addEventListener("click", function () {
+  dropdList.remove(dropdList.selectIndex);
+});
 
```

Comment:
the event listener removes first item option when it clicked instead of selected option. selecting dropList option element each time when button clicked may solve the problem 

dropdList.id = "select_colors";
//function for removing the selected option
removeBtn.addEventListener("click", function () {
  var x = document.getElementById("select_colors");
  x.remove(x.selectedIndex);  
});

Reply:
Thanks Sinan for pointing to this flaw in the code. I was surprised since initially everything was fine.  With you help I realized that at some point of time I lost somehow suffix "ed". It is strange that though there is no such property as "selectIndex" VS code debugging console didn't warn about it. I just changed this.

Reply:
Great , Thanks for letting me know ;)